### PR TITLE
[AUTO][Benchmark APP] sync config parsing logic of benchmark python version with benchmark c++ version

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - sync config parsing logic of benchmark python version with benchmark C++ version

### Tickets:
 - CVS-94329

### CODE REVIEW

**Review by nkogteva — APPROVED** [review]

File: tools/benchmark_tool/openvino/tools/benchmark/utils/utils.py [hunk 1]
```diff
@@ -723,10 +723,47 @@ def show_available_devices():
...
+            continue
+        for property_name in config[device]:
+            property_value = config[device][property_name]
+            if property_name in supported_properties:
+                properties[device][property_name] = property_value
+            else:
+                if 'DEVICE_PROPERTIES' not in properties[device]:
+                    properties[device]['DEVICE_PROPERTIES'] = {}
+                if property_name not in properties[device]['DEVICE_PROPERTIES']:
+                    properties[device]['DEVICE_PROPERTIES'][property_name] = {}
```
Comment (Wovchena) [inline tools/benchmark_tool/openvino/tools/benchmark/utils/utils.py:742]:
```suggestion
                properties[device].setdefault('DEVICE_PROPERTIES', {})
                properties[device]['DEVICE_PROPERTIES'].setdefault(property_name, {})
```

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: tools/benchmark_tool/openvino/tools/benchmark/parameters.py [hunk 2]
```diff
@@ -24,9 +24,17 @@ def __call__(self, parser, namespace, values, option_string=None):
...
+          lines = super()._split_lines(text, width)
+          lines += ['']
+          if "simple JSON file" not in text:
+               return lines
+          lines = text.split('\n')
+          return lines
 
 def parse_args():
-    parser = argparse.ArgumentParser(add_help=False)
+    parser = argparse.ArgumentParser(add_help=False, formatter_class=HelpFormatterWithLines)
```
[the remark this answers is not in the document]
Comment (yangwang201911) [reply tools/benchmark_tool/openvino/tools/benchmark/parameters.py:37]: This change replacing `HelpFormatterWithLines` with `RawTextHelpFormatter`, will affect the display format of other help msg. The display length of help msg using `RawTextHelpFormatter` is too long and not aligned, while `HelpFormatterWithLines` only changes the help msg display format of `load_config`. Need your confirm.

**Review by wangleis — APPROVED** [review]

File: tools/benchmark_tool/openvino/tools/benchmark/parameters.py [hunk 3]
```diff
@@ -129,8 +137,29 @@ def parse_args():
...
-                           " Please note, command line parameters have higher priority then parameters from configuration file.")
+                      help="Optional. Path to JSON file to load custom OpenVINO parameters.\n"
+                           "Please note, command line parameters have higher priority then parameters from configuration file.\n"
+                           "Example 1: a simple JSON file for HW device with primary properties.\n"
+                           "             {\n"
+                           "                \"CPU\": {\"NUM_STREAMS\": \"3\", \"PERF_COUNT\": \"NO\"}\n"
+                           "             }\n"
+                           "Example 2: a simple JSON file for meta device(AUTO/MULTI) with HW device properties.\n"
+                           "             {\n"
+                           "             	\"AUTO\": {\n"
```
[the remark this answers is not in the document]
Comment (yangwang201911) [reply tools/benchmark_tool/openvino/tools/benchmark/parameters.py]: Done

File: tools/benchmark_tool/openvino/tools/benchmark/utils/utils.py [hunk 4]
```diff
@@ -723,10 +723,47 @@ def show_available_devices():
...
+                for key in properties_dict:
+                    properties[device]['DEVICE_PROPERTIES'][property_name][key] = properties_dict[key]
+
     with open(filename, 'w') as f:
-        json.dump(config, f, indent=4)
+        json.dump(properties, f, indent=4)
 
 
 def load_config(filename, config):
+    original_config = {}
```
[the remark this answers is not in the document]
Comment (yangwang201911) [reply tools/benchmark_tool/openvino/tools/benchmark/utils/utils.py]: Done

File: tools/benchmark_tool/openvino/tools/benchmark/utils/utils.py [hunk 5]
```diff
@@ -723,10 +723,47 @@ def show_available_devices():
 
 
 def dump_config(filename, config):
+    properties = {}
+    for device in config:
+        properties[device] = {}
+        supported_properties = Core().get_property(device, 'SUPPORTED_PROPERTIES')
+        # check if ov::device::properties exists in the config
+        if device not in [AUTO_DEVICE_NAME, MULTI_DEVICE_NAME]:
```
[the remark this answers is not in the document]
Comment (yangwang201911) [reply tools/benchmark_tool/openvino/tools/benchmark/utils/utils.py]: Done

File: tools/benchmark_tool/openvino/tools/benchmark/utils/utils.py (hunk 1 above)
Comment (yangwang201911) [reply tools/benchmark_tool/openvino/tools/benchmark/utils/utils.py]: Done

File: tools/benchmark_tool/openvino/tools/benchmark/parameters.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Wovchena) [reply tools/benchmark_tool/openvino/tools/benchmark/parameters.py:37]: Well, that would align C++ and Python implementations even more. C++ also doesn't limit message length. But I will approve, not to block you

**Pull request merged**